clients/pkg/promtail/client/manager.go

```diff
@@ -0,0 +1,49 @@
+package client
+
+import (
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/wal"
+)
+
+// Manager manages remote write client instantiation, and connects the related components to orchestrate the flow of api.Entry
+// from the scrape targets, to the remote write clients themselves.
+//
+// Right now it just supports instantiating the WAL writer side of the future-to-be WAL enabled client. In follow-up
+// work, tracked in https://github.com/grafana/loki/issues/8197, this Manager will be responsible for instantiating all client
+// types: Logger, Multi and WAL.
+type Manager struct {
+	clientChannelHandler api.EntryHandler
+}
```

Comment:
We want each steps PR to be as standalone as possible and ideally not introduce a codepath that does nothing, let me know if this makes sense:
- in this PR the Manager should also contain a field `clients []Client`
- manager should also implement the Client interface, and it's functions would just range over it's internal `clients`

This would allow us to instantiate the Manager and any current implementation of the `Client` interface, and still be able to write both to the WAL and to the client correctly and thus "enable" the WAL but not break remote writing of log lines.